packages/service-proxy/src/mixins/service.mixin.ts

```diff
@@ -0,0 +1,198 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/service-proxy
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {Provider} from '@loopback/context';
+import {Application} from '@loopback/core';
+
+/**
+ * Interface for classes with `new` operator.
+ */
+export interface Class<T> {
```

Comment:
Should we somehow leverage the `Class` type from https://github.com/strongloop/loopback-next/blob/master/packages/context/src/value-promise.ts#L13 or refactor it into a common package?

Reply:
Having a shared definition of `Class` and `Constructor` types would be definitely nice.

However, I am not sure if such small typedefs are enough to justify the cost of creating a new package?

Also please note that in `RepositoryMixin` (which I based my work on), `Class` is coming from a different package - see https://github.com/strongloop/loopback-next/blob/f68a45ced9a392286d484b72cc690a6599aeb0c0/packages/repository/src/common-types.ts#L11-L19

Can we leave this improvement out of scope of this pull request please?

Reply:
Sure. +1 to have a separate PR.